cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -199,17 +199,54 @@ func generateMountsModeRun() error {
 	if err != nil {
 		return err
 	}
-	// 2.2 mount base
+
+	// 2.2.1 check base status and try base if we are updating the base snap
+	baseSnapPath := ""
+	switch modeEnv.BaseStatus {
+	case "try":
+		// then we are trying a base snap update and there should be a try_base
+		// set in the modeenv too
+		if modeEnv.TryBase != "" {
+			// check that the TryBase exists in ubuntu-data
+			tryBaseSnapPath := filepath.Join(dataDir, "system-data", dirs.SnapBlobDir, modeEnv.TryBase)
+			if osutil.FileExists(tryBaseSnapPath) {
+				// set the TryBase and mount it
+				// TODO:UC20: should we write the modeenv _first_ here then
+				// output to the initramfs to create this mount for maximum
+				// reboot resiliency?
+				modeEnv.BaseStatus = "trying"
+				baseSnapPath = tryBaseSnapPath
+			} else {
+				return fmt.Errorf("try base snap %q does not exist", modeEnv.TryBase)
+			}
+		} else {
+			// TODO:UC20: should we log a message here about missing the
+			// try_base?
+			return fmt.Errorf("base status is try but no try base is set")
+		}
+	case "trying":
```

Comment:
thinking about this, should we also re-set the modeenv BaseStatus here like we do in the grub bootloader on a failed base snap? See https://github.com/snapcore/pc-amd64-gadget/blob/d081821c25b0e36c9b98c3f500e7e82f512a3b4b/grub.cfg-boot#L21-L22

Reply:
also I'm missing a unit test for this fallback case, will add shortly